plugins/factory-plugin/src/git.ts

```diff
@@ -0,0 +1,77 @@
+/*********************************************************************
+ * Copyright (c) 2019 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ **********************************************************************/
+const { spawn } = require('child_process');
+
+export interface GitUpstreamBranch {
+    remote: string;
+    branch: string;
+    remoteURL?: string;
+}
+
+export async function getUpstreamBranch(projectPath: string): Promise<GitUpstreamBranch | undefined> {
+
+    return new Promise<GitUpstreamBranch | undefined>((resolve, reject) => {
+        const gitRevParseCmd = spawn('git', ['rev-parse', '--abbrev-ref', '--symbolic-full-name', '@{upstream}'], { cwd: projectPath });
```

Comment:
would be nice if we could use https://github.com/sindresorhus/execa
```
const execa = require('execa');
const {stdout} = await execa('echo', ['unicorns']);
```

Reply:
but it requires CQ probably

Reply:
probably :)